dask/dataframe/core.py

```diff
@@ -1654,6 +1661,44 @@ def count(self, axis=None, split_every=False):
                 result.divisions = (self.columns.min(), self.columns.max())
             return result
 
+    @derived_from(pd.DataFrame)
+    def mode(self, axis=0, dropna=True, split_every=False):  # this is the series class
+        """Use a counter on each partition, sum the resulting counters, iterate over key,
+        value pairs until num_occurrences isn't equal to the first one"""
+        if axis == 1:
+            meta = self._meta_nonempty.count(axis=axis)
+            result = map_partitions(
+                M.mode,
+                self,
+                meta=meta,
+                token=self._token_prefix + "mode",
+                axis=axis,
+                dropna=True,
+                enforce_metadata=False,
+            )
+            return result
+        else:
+            from collections import Counter
+
+            def _mode_chunk(df, dropna=dropna):  # already defined elsewhere now
+                if dropna:
+                    return Counter(df.dropna().values)
+                else:
+                    return Counter(df.values)
+
+            all_values_dict = self.reduction(
+                chunk=_mode_chunk,
+                aggregate=M.sum,
+                meta="object",
+                split_every=split_every,
+            ).to_delayed()
+            # now I have the all_values_dict as a delayed object.  Next, find max value and then all keys with max value
+            max_val = delayed(lambda x: max(x.values()))(all_values_dict)
+            mode_list = delayed(
+                lambda x, max_val: sorted([k for k, v in x.items() if v == max_val])
+            )(all_values_dict, max_val)
+            return mode_list
```

Comment:
What's the return type here? This should return a DataFrame I think.

Reply:
No longer applicable